Library/Homebrew/cask/artifact/abstract_uninstall.rb

```diff
@@ -396,7 +396,10 @@ def uninstall_rmdir(*directories, command: nil, **_)
               command.run!("/bin/rm", args: ["-f", "--", ds_store], sudo: true, print_stderr: false)
             end
 
-            command.run("/bin/rmdir", args: ["--", resolved_path], sudo: true, print_stderr: false)
+            uninstall_rmdir(*resolved_path.children, command: command) if resolved_path.children.all?(&:directory?)
+
+            status = command.run("/bin/rmdir", args: ["--", resolved_path], sudo: true, print_stderr: false)
+            next unless status.success?
```

Comment:
This doesn't do anything.

Reply:
> This doesn't do anything.

@reitermarkus The functionality is working… The tests pass, and I've tested it independently with a test Cask.

> Maybe add `.success?` so we can return early if the recursive call returns `false`.

Where did you mean `.success?` should go exactly? This line was originally `return unless`, but rubocop complained…

Reply:
Basically it should return early if any recursive `rmdir` fails.

Reply:
Ok, so I had to return `nil` to stop rubocop complaining…
~~~ rb
return nil unless status.success?
~~~
Tested and still working… does this do the job @reitermarkus?